src/Components/Menu/ModalMenu.js

```diff
@@ -0,0 +1,141 @@
+import React, {useState} from 'react';
+import { Modal, Col, Row } from 'antd';
+import Button from '../Button/Button';
+import shortid from 'shortid';
+import Extras from '../Menu/Extras'
+
+
+const ModalMenu = ({item, carrito, addProducto}) => {
+  
+  const extrasList = [
+    "Queso",
+    "Pepperoni",
+    "Tocino",
+    "Huevo"
+  ];
+
+  function getDefaultExtras() {
+    return extrasList.map(extra => ({
+        ingrediente: extra,
+        checked: false
+    }))
+}
+
+  const quantity = useQuantity(carrito.item && carrito.item.quantity)
+  const extras = useExtras(carrito.item.extras)
+
+  function useExtras(defaultExtra){
+    
+    const [ extras, setExtras] = useState(
+      defaultExtra || getDefaultExtras()
+    )
+    
+    function checkExtra(i){
+        const newExtras = [...extras];
+        newExtras[i] = { ...newExtras[i], checked: !newExtras[i].checked}
+        setExtras(newExtras)
+    }
+
+    function resetExtras() {
+      setExtras(getDefaultExtras())
+    }
+   
+    return {
+        checkExtra,
+        extras,
+       resetExtras
+    }
+}
+
+  function useQuantity(defaultQuantity) {
+    const [value, setvalue] = useState(defaultQuantity || 1);
```

Comment:
esta muy general el nombre de esta variable
```suggestion
    const [quantityValue, setQuantityValue] = useState(defaultQuantity || 1);
```